crates/fuel-core/src/database/block.rs

```diff
@@ -115,18 +115,18 @@ impl StorageMutate<FuelBlocks> for Database {
             .next()
             .transpose()?
             .map(|(_, metadata)| metadata)
-            .unwrap_or_default();
+            .unwrap_or(DenseMerkleMetadataV1::default().into());
 
         let storage = self.borrow_mut();
         let mut tree: MerkleTree<FuelBlockMerkleData, _> =
-            MerkleTree::load(storage, prev_metadata.version)
+            MerkleTree::load(storage, prev_metadata.version())
                 .map_err(|err| StorageError::Other(anyhow::anyhow!(err)))?;
         tree.push(block_id.as_slice())?;
 
         // Generate new metadata for the updated tree
         let version = tree.leaves_count();
         let root = tree.root();
-        let metadata = DenseMerkleMetadata { version, root };
+        let metadata = DenseMerkleMetadataV1 { version, root }.into();
```

Comment:
The same here, we can have a basic constructor since these fields will be used in most cases